Src/WindowsServer/WindowsServer.Shared/ConfigurationTelemetryInitializer.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    /// <summary>
+    /// A telemetry initializer that will gather configuration information.
+    /// </summary>
+    public class ConfigurationTelemetryInitializer : ITelemetryInitializer
+    {
+        private string roleName;
+        private string roleInstanceName;
+        private volatile bool updateEnvVars = true;
```

Comment:
`volatile` is probably overkill since there's no interlocking on this value.
Also, suggest reversing the 'polarity' so that it can be initialized to `false` (which is the default) and set to `true` when values have been read.

Reply:
Totally agree. I just followed the implementation of [`AzureWebAppRoleEnvironmentTelemetryInitializer`](https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs#L26).